### ISSUE TITLE
Investigate reworking release workflow

### ISSUE THREAD

**Issue by mnriem** [piranhacloud/piranha#2697]
Investigate if we can use the latest Java to create the release so we can push out bits that require Java versions post Java 17 while still making sure the core of Piranha uses Java 17

**Comment by mnriem** [comment]
@Thihup If you have some bandwidth can you do some dry run testing? (aka simulate using the steps in the release workflow). I would love to see the Virtual Threads module to be released :)

**Comment by Thihup** [comment]
Right now this is blocked by #2642

**Comment by mnriem** [comment]
OK noted. Hopefully it can be resolved soon

**Comment by mnriem** [comment]
Closing in favor of #2832
